File: Data Structures & Algorithms/pow-x-n/submission-0.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        def rec(x,t):
            if t==0:
                return 1;
            if t==1:
                return x;
            temp=rec(x,t//2)*rec(x,t//2)
            if t%2==0:
                return temp
            else:
                return temp*x
        if n<0:
            return 1/rec(x,-n)
        return rec(x,n)

File: Data Structures & Algorithms/reverse-integer/submission-0.cpp
class Solution {
public:
    int reverse(int x) {
        int ans=0;
        int temp=x;
        int i=0;
        while(temp){
            int cur=temp%10;
            if(ans<INT_MIN/10 || ans>INT_MAX/10){
                return 0;
            }else if((ans==INT_MAX/10 && cur>INT_MAX%10)||(ans==INT_MIN/10 && cur<INT_MIN%10)){
                return 0;
            }
            else{
                ans=ans*10+cur;
            }
            temp=temp/10;
            i++;
        }
        return ans;
    }
};
